### ISSUE TITLE
Options Menu Streamlining Suggestions

### ISSUE THREAD

**Issue by OpenRift412** [fabiangreffrath/woof#1854]
So I've had a few more ideas for how we can streamline just a few things and maybe cut out some superfluous toggles:
- merge the "fullscreen" and "exclusive fullscreen toggles", allowing the fullscreen options to be "off", "on", and "exclusive"
- maybe remove the "default skill" option, as I really can't see it being all that useful for most, especially with a mouse-driven menu. maybe for those UV-only die-hards, they can just edit the config file themselves if they really need it.
- maybe merge the "uncapped framerate" and "framerate limit" options. this could be done in one of two ways:
  1. Have the user select from a few fps cap presets: 35, 60, 90, 120, 144, 165, 200, 240, and Uncapped
  2. Just have the "framerate limit" option where you enter the number. if set to 35, disable interpolation and run at 35 FPS. If anything 
  higher, enable interpolation and run at desire framerate. If set to 0, run uncapped with interpolation, and the option will display as 
  "uncapped" upon hitting enter.

**Comment by ceski-1** [comment]
> merge the "fullscreen" and "exclusive fullscreen toggles", allowing the fullscreen options to be "off", "on", and "exclusive"

This was considered before and the [conclusion at the time](https://github.com/fabiangreffrath/woof/pull/1402#discussion_r1468929862) was that exclusive fullscreen can have issues with certain configurations. However, I think after the [fixes added by this PR](https://github.com/fabiangreffrath/woof/pull/1754), it's worth revisiting this idea again. I agree that it makes more sense combined into one setting.

> maybe remove the "default skill" option, as I really can't see it being all that useful for most, especially with a mouse-driven menu. maybe for those UV-only die-hards, they can just edit the config file themselves if they really need it.

I don't have an opinion on this.

> maybe merge the "uncapped framerate" and "framerate limit" options. this could be done in one of two ways

[This was also considered before](https://github.com/fabiangreffrath/woof/pull/1451) and there were concerns it would introduce confusion in other ways. But I agree that it still feels like there's a better solution somewhere.

> 35, 60, 90, 120, 144, 165, 200, 240, and Uncapped

While convenient, whole numbers that match the monitor's refresh rate are [not ideal for input latency](https://blurbusters.com/gsync/gsync101-input-lag-tests-and-settings/14/). This can be verified with [PresentMon](https://github.com/GameTechDev/PresentMon), MSI Afterburner, etc.

**Comment by rfomin** [comment]
> > merge the "fullscreen" and "exclusive fullscreen toggles", allowing the fullscreen options to be "off", "on", and "exclusive"
> 
> This was considered before and the [conclusion at the time](https://github.com/fabiangreffrath/woof/pull/1402#discussion_r1468929862) was that exclusive fullscreen can have issues with certain configurations. However, I think after the [fixes added by this PR](https://github.com/fabiangreffrath/woof/pull/1754), it's worth revisiting this idea again. I agree that it makes more sense combined into one setting.

Better than before, but still weird on my old Windows 10 system. I think it should be a separate “experts only” option.

> maybe remove the "default skill" option, as I really can't see it being all that useful for most, especially with a mouse-driven menu. maybe for those UV-only die-hards, they can just edit the config file themselves if they really need it.

I'm one of those UV-only players 😄 This was a feature request a long time ago. The old MBF.EXE just remembers the last used skill.

**Comment by ceski-1** [comment]
> Better than before, but still weird on my old Windows 10 system. I think it should be a separate “experts only” option.

Sure, no problem. I think the next time to revisit the topic would be whenever SDL3 migration occurs. Direct3D11 will be the default and exclusive fullscreen may no longer be needed at all.

**Comment by OpenRift412** [comment]
> I'm one of those UV-only players 😄 This was a feature request a long time ago. The old MBF.EXE just remembers the last used skill.

Well so am I, but it's just such a redundant feature that really doesn't need to take up menu space in my opinion. Why stop there? Why not have a default main menu item toggle? It just seems like inconsistent design to have such a specific setting like this. But I suppose if this was something somebody wanted, I suppose it's whatever.

**Comment by fabiangreffrath** [comment]
> * maybe remove the "default skill" option, as I really can't see it being all that useful for most, especially with a mouse-driven menu. maybe for those UV-only die-hards, they can just edit the config file themselves if they really need it.

Although it's an option that's usually only changed once, I still think this is an important option to have. I'd find it sad if it would get burried in the config file.

**Comment by OpenRift412** [comment]
So I have another suggestion, similar to what I was suggesting with the fullscreen toggles. Would it be possible to merge the "Free Look" and "Invert Look" toggles, having "Free Look" toggle between "Off", "On", and "Inverted"? Or is that a similar situation to the fullscreen merge suggestion from before?

**Comment by rfomin** [comment]
> Would it be possible to merge the "Free Look" and "Invert Look" toggles, having "Free Look" toggle between "Off", "On", and "Inverted"? Or is that a similar situation to the fullscreen merge suggestion from before?

No, the fullscreen situation is unique because toggling to “Exclusive Fulscreen” looks glitchy on many systems, and we don't want it to happen by accident. Perhaps “Exclusive Fulscreen” won't be needed in SDL3, and we'll just remove it.

Sure, we can merge the "Free Look" and "Invert Look", but we are not so much pressed for space on the "Mouse" tab.

**Comment by gendlin** [comment]
As someone who plays inverted I'm used to skimming through options menus in search of the string "invert(ed)" - would prefer if this was not hidden away somewhere making it harder to find at a quick glance

**Comment by OpenRift412** [comment]
> As someone who plays inverted I'm used to skimming through options menus in search of the string "invert(ed)" - would prefer if this was not hidden away somewhere making it harder to find at a quick glance

That's a good point. I could go on about how it's kind of pointless and broken vertical look in Doom is but that's not really important.

**Comment by fabiangreffrath** [comment]
We need to get rid of the "Translucency Filter" and "Gray Percent Sign" menu options.

**Comment by elf-alchemist** [comment]
Perhaps merge Gray Percent into Colored Hud setting? Something akin to:
* Boom Colored HUD: "Off", "Colored Numbers", "Gray Percent Sign", "Both"


As for translucency of course it should be kept in `woof.cfg`, but I was thinking maybe a command line option could be added that takes precedence over the config file setting, purely as a more helpful tool in generating `TRANMAP`s

**Comment by OpenRift412** [comment]
> We need to get rid of the "Translucency Filter" and "Gray Percent Sign" menu options.

I agree.
